job_definitions/update_api_tokens.yml

```diff
@@ -0,0 +1,74 @@
+{% set stages = [{'name': 'Add new tokens', 'command': 'add-new'}, {'name': 'Remove old tokens', 'command': 'remove-old'}] %}
+{% set app_groups = [['search-api'], ['api'], ['admin-frontend', 'briefs-frontend', 'brief-responses-frontend', 'buyer-frontend', 'supplier-frontend', 'user-frontend']] %}
+---
+- job:
+    name: update-api-tokens
+    display-name: "Update API tokens"
+    project-type: pipeline
+    descript1ion: Update Data and Search API tokens for a given environment.
+    disabled: false
+    concurrent: false
+    parameters:
+      - choice:
+          name: STAGE
+          choices:
+            - preview
+            - staging
+            - production
+    pipeline:
+      script: |
+        def notify_slack(icon, status) {
+          build job: "notify-slack",
+          parameters: [
+            string(name: 'USERNAME', value: "update_api_tokens"),
+            string(name: 'ICON', value: icon),
+            string(name: 'JOB', value: "Update API tokens on ${STAGE}"),
+            string(name: 'CHANNEL', value: "#dm-release"),
+            text(name: 'STAGE', value: "${STAGE}"),
+            text(name: 'STATUS', value: status),
+            text(name: 'URL', value: "<${BUILD_URL}consoleFull|${BUILD_DISPLAY_NAME}>")
+          ]
+        }
+
+{% for stage in stages %}
+        stage("{{ stage.name }}") {
+{% if loop.index == 1 %}
+          currentBuild.displayName = "#${BUILD_NUMBER} - ${STAGE}"
+{% endif %}
+          node {
+            try {
+              sh('docker run -e GITHUB_ACCESS_TOKEN digitalmarketplace/scripts scripts/update-api-tokens.sh "{{ stage.command }}" "${STAGE}"')
```

Comment:
Where does `GITHUB_ACCESS_TOKEN` come from? If it's a variable should it be `"$GITHUB_ACCESS_TOKEN"`?

Reply:
It's an environment variable. When passing via the `-e` flag of docker run it will look for an env variable called that on the host and inject it, along with its value, into the container. We do this so we don't have to show the token itself in the command.

Reply:
So it's an env variable on the Jenkins box? I can only find `GITHUB_TOKEN` in the jenkins vars in the creds repo? And it'll still need to be `${GITHUB_TOKEN}`, no?